common/sgx/eeid_verifier.c

```diff
@@ -317,36 +379,23 @@ static oe_result_t _eeid_verify_evidence(
             &sgx_claims_length,
             &parsed_report));
 
+        oe_free_claims(sgx_claims, sgx_claims_length);
```

Comment:
If you don't care about sgx_claims output, you may pass in NULL. `_verify_sgx_report()` may ignore extracting SGX claims if the caller is not interested it.

Reply:
I thought so too, but that's not the case. `oe_sgx_extract_claims` assumes that `claims_out` is non-NULL and (my) `_verify_sgx_report` doesn't free its local claims, which would result in a memory leak. I'll move the `oe_free_claims` there. I'm happy to fix `oe_sgx_extract_claims` though, if that's welcome. I found it odd that in this particular case we're not allowed to pass NULL.

Reply:
I think that `oe_sgx_extract_claims()` should always expect a non-NULL `claims_out` parameter, since the purpose of this function is to extract claims. But `_verify_sgx_report()` should check its `sgx_claims` parameter, and skip calling `oe_sgx_extract_claims()` if this parameter is NULL.

Similarly, `_verify_evidence()` in `common/sgx/verifier.c` should check its `claims` parameter, and skip calling `oe_sgx_extract_claims()` if this parameter is NULL.

On the other hand, in `include/openenclave/attestation/verifier.h`, the `claims` parameter in API `oe_verify_evidence()` is not commented as optional, but the implementation of this API in `common/attest_plugin.c` allows this parameter to be NULL.

So looks like there is some clean up in comments and implementation. How about taking care of this in a separate issue / PR? @yentsanglee @wintersteiger

Reply:
That's a very good point indeed! I just copied the structure from the SGX plugin, but I ended up using only the parsed report. I should really change and get the information from the claims instead, which fits in much better with the plugin infrastructure.

Reply:
(Done)